packages/protocol/contracts/governance/DowntimeSlasher.sol

```diff
@@ -35,99 +53,198 @@ contract DowntimeSlasher is SlasherUtil {
   }
 
   /**
-   * @notice Sets the slashable downtime
+   * @notice Sets the slashable downtime.
    * @param interval Slashable downtime in blocks.
    */
   function setSlashableDowntime(uint256 interval) public onlyOwner {
-    require(interval != 0, "Slashable downtime cannot be zero");
-    require(interval < getEpochSize(), "Slashable downtime must be smaller than epoch size");
+    require(interval != 0, "slashable downtime cannot be zero");
     slashableDowntime = interval;
     emit SlashableDowntimeSet(interval);
   }
 
   /**
-   * @notice Test if a validator has been down.
-   * @param startBlock First block of the downtime. Last block will be computed from this.
-   * @param startSignerIndex Index of the signer within the validator set as of the start block.
-   * @param endSignerIndex Index of the signer within the validator set as of the end block.
-   * @return True if the validator signature does not appear in any block within the window.
-   * @dev Due to getParentSealBitmap, startBlock must be within 4 epochs of the current head.
+   * @notice Calculates and returns the signature bitmap for the specified interval.
+   * This bitmap will contain a one for any validator that signed at least one block in that
+   * interval, and zero otherwise.
+   * @param startBlock First block of the interval.
+   * @param endBlock Last block of the interval.
+   * @return The signature uptime bitmap for the specified interval.
+   * @dev startBlock and endBlock must be in the same epoch.
+   * @dev The getParentSealBitmap precompile requires that startBlock must be within 4 epochs of 
+   * the current block.
    */
-  function isDown(uint256 startBlock, uint256 startSignerIndex, uint256 endSignerIndex)
+  function getBitmapForInterval(uint256 startBlock, uint256 endBlock)
     public
     view
-    returns (bool)
+    returns (bytes32)
   {
-    uint256 endBlock = getEndBlock(startBlock);
-    require(endBlock < block.number.sub(1), "end block must be smaller than current block");
+    require(endBlock >= startBlock, "endBlock must be greater or equal than startBlock");
+    // The signature bitmap for block N is stored in block N+1.
+    // The latest block is `block.number - 1`, which stores the signature bitmap for
+    // `block.number - 2`.
+    uint256 lastBlockWithSignatureBitmap = block.number.sub(2);
+    require(
+      endBlock <= lastBlockWithSignatureBitmap,
+      "the signature bitmap for endBlock is not yet available"
+    );
+    uint256 epochSize = getEpochSize();
     require(
-      startSignerIndex < numberValidatorsInSet(startBlock),
-      "Bad validator index at start block"
+      lastBlockWithSignatureBitmap.sub(startBlock) <= epochSize.mul(4),
```

Comment:
This should be `block.number.sub(startBlock) < epochSize.mul(4)` to match the implementation https://github.com/celo-org/celo-blockchain/blob/08d083b78877f9558f467e262cd62a2bbb86afb9/core/vm/contracts.go#L1009-L1013
```suggestion
      block.number.sub(startBlock) < epochSize.mul(4),
```